components/script/dom/node.rs

```diff
@@ -604,6 +604,10 @@ impl Node {
         }
     }
 
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
```

Comment:
The point of `is_empty` is that it can be faster than `len()`. But `CharacterData`'s `Length` iterates the string (see https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#264-266).

So if you add this, you should implement this in a faster way.

Reply:
Got it!